### PR TITLE
recreated PR fullscreen in GridPaint activity

### LINKED ISSUE #601: Add a fullscreen button in the GridPaint activity
Sugarizer activities don't look good on devices with small screen - for example smartphones.
On these devices, the toolbar could sometimes take 1/4 of the screen.

To increase the size for activities contents, the idea is to add a fullscreen button in the toolbar  like in the QRCode activity. When clicked, the fullscreen button hide the toolbar and display only an unfullscreen button. When clicked, the unfullscreen button show again the toolbar.

This features is implemented here: https://github.com/llaske/sugarizer/blob/dev/activities/QRCode.activity/js/activity.js#L144-L160

The objective of this issue is to implement the same feature in the GridPaint activity.

Thinks to check:

* [ ] The fullscreen button should be on the right of the toolbar, near the Stop button.
* [ ] The fullscreen button should be always visible, including in small screen size. Could require to adapt media queries in CSS.
* [ ] The size of the board should be increase in fullscreen mode. So it means that the resize event should be call directly or indirectly when clicking on fullscreen/unfullscreen button.
* [ ] The unfullscreen button should be visible only when the screen is in fullscreen mode.
* [ ] The unfullscreen button should be attach to the right of the screen. It's better if it's not at the same location than the Stop button to avoid side effect on multiple click.

### PR DESCRIPTION
**Pull request by Swayamsvk**
fixed #601 
grid3 mkv

### CODE REVIEW

**Comment by Swayamsvk** [comment]
Actually I recreated the previous PR in which mixing of commits was taking place.From here on I will work on updations.

File: activities/Gridpaint.activity/js/activity.js [hunk 1]
```diff
@@ -63,6 +63,22 @@ define(["sugar-web/activity/activity","sugar-web/datastore","sugar-web/env","web
...
 			tutorial.start();
 		});
-    });
+
+			// Full screen
+		document.getElementById("fullscreen-button").addEventListener('click', function() {
+			document.getElementById("main-toolbar").style.display = "none";
+			document.getElementById("canvas").style.top = "0px";
+			document.getElementById("unfullscreen-button").style.visibility = "visible";
+
```
Comment (llaske) [inline activities/Gridpaint.activity/js/activity.js]: You should call screen resize to allow the activity to use the free space.

File: activities/Gridpaint.activity/js/activity.js [hunk 2]
```diff
@@ -63,6 +63,22 @@ define(["sugar-web/activity/activity","sugar-web/datastore","sugar-web/env","web
...
+			document.getElementById("main-toolbar").style.display = "none";
+			document.getElementById("canvas").style.top = "0px";
+			document.getElementById("unfullscreen-button").style.visibility = "visible";
+			
+		});
+		document.getElementById("unfullscreen-button").addEventListener('click', function() {
+			document.getElementById("main-toolbar").style.display = "block";
+			document.getElementById("canvas").style.top = "55px";
+			document.getElementById("unfullscreen-button").style.visibility = "hidden";
+
```
Comment (llaske) [inline activities/Gridpaint.activity/js/activity.js]: Same here.

File: activities/Gridpaint.activity/css/activity.css [hunk 3]
```diff
@@ -137,4 +137,22 @@
...
+
+#unfullscreen-button {
+  background-image: url(../icons/view-return.svg);
+  border: 0;
+  border-radius: 0px;
+  margin:1px;
+  width: 45px;
+  height: 45px;
+  top: 0;
+  right: 0;
```
Comment (llaske) [inline activities/Gridpaint.activity/css/activity.css]: Should be 5% instead of 0

**Comment by Swayamsvk** [comment]
I will make the other changes soon.

**Comment by Swayamsvk** [comment]
I tried this function computesize as you have mentioned. It helps me to resize the canvas but I am not able to remove the black margin which is appearing on the bottom of the canvas.
Screenshot from 2020-02-20 20-56-48

**Comment by llaske** [comment]
Hard to help without seeing your source code...
By the way, did you update [this line](https://github.com/llaske/sugarizer/blob/dev/activities/Gridpaint.activity/js/events.js#L47) to handle the fact that the toolbar could be hidden so its size will become 0 not 55?

**Comment by Swayamsvk** [comment]
I tried it but it's not helping. Actually, I was looking at other similar PRs such as #607 and I have noticed that they are not including anything else other than the listeners. What could be the reason that they are not having the same problem?

**Comment by llaske** [comment]
Once again, without seeing your code, I can't help.
Please update your PR with your last try.

**Comment by Swayamsvk** [comment]
Resolved the appearance of the black margin at the bottom.

File: activities/Gridpaint.activity/js/activity.js [hunk 4]
```diff
@@ -63,6 +65,42 @@ define(["sugar-web/activity/activity","sugar-web/datastore","sugar-web/env","web
 		document.getElementById("help-button").addEventListener('click', function(e) {
 			tutorial.start();
 		});
-    });
+
+		
+
+		function computeSize() {
```
Comment (llaske) [inline activities/Gridpaint.activity/js/activity.js]: You're not supposed to rewrite this function. It's already a global function.

File: activities/Gridpaint.activity/js/activity.js [hunk 5]
```diff
@@ -63,6 +65,42 @@ define(["sugar-web/activity/activity","sugar-web/datastore","sugar-web/env","web
...
+				//manipulates size of main frame imp here
+				element.style.width = "1024px";
+				element.style.height = "748px";
+			}
+			setTransform(document.getElementById("gridcnv"));
+			setTransform(document.getElementById("hitbuffer"));
+			setTransform(document.getElementById("thumbframe"));
+		}
+		
+        // Full screen
```
Comment (llaske) [inline activities/Gridpaint.activity/js/activity.js]: Don't mix indentation style.

File: activities/Gridpaint.activity/index.html [hunk 6]
```diff
@@ -31,9 +31,11 @@
   <body style="overflow-x: hidden;">
     <div id="main-toolbar" class="toolbar">
       <button class="toolbutton" id="activity-button" title="Grid Paint Activity"></button>
-			<button class="toolbutton" id="clear-button" title="Clear" style="display: none;"></button>
+	  <button class="toolbutton" id="clear-button" title="Clear" style="display: none;"></button>
```
Comment (llaske) [inline activities/Gridpaint.activity/index.html]: Avoid changing line when nothing is changed.

File: activities/Gridpaint.activity/index.html [hunk 7]
```diff
@@ -31,9 +31,11 @@
...
       <button class="toolbutton pull-right" id="stop-button" title="Stop"></button>
-      <button class="toolbutton pull-right" id="help-button" title="Tutorial"></button>
+	  <button class="toolbutton pull-right" id="help-button" title="Tutorial"></button>
```
Comment (llaske) [inline activities/Gridpaint.activity/index.html]: Avoid changing line when nothing is changed.

**Comment by Swayamsvk** [comment]
@llaske actually I made a terrible mistake. While I was working on this issue I realized I forked the master branch, not the dev branch that's why I was not able to get the function defined on the dev branch. But somehow I created the dev branch in my system with the current changes. Should I work on it or I should fork the current dev branch separately. Please let me know what will be better.

**Comment by llaske** [comment]
@Swayamsvk I suggest you backup updated files and recreate a branch from the dev branch. I will be easier.

**Comment by Swayamsvk** [comment]
I have made the necessary changes required. Please let me know if anything else is required.

**Comment by llaske** [comment]
* Clicking on fullscreen doesn't change the size of the screen
* The unfullscreen button is out of the screen 

Peek 07-03-2020 15-29

* There is two error messages in the console:

```
activity.js:75 Uncaught ReferenceError: events is not defined
    at HTMLButtonElement.<anonymous> (activity.js:75)
(anonymous) @ activity.js:75
activity.js:81 Uncaught ReferenceError: events is not defined
    at HTMLButtonElement.<anonymous> (activity.js:81)
(anonymous) @ activity.js:81
```

**Comment by llaske** [comment]
Fixed in https://github.com/llaske/sugarizer/commit/fdfac92b2711b62a4777e7e0be2479228c228abc

**Pull request closed without merging**